### PR TITLE
Update "Define a Function" and "External Dependencies"

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-14954
https://jira.mongodb.org/browse/DOCSP-14955

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/cht_cli/functions/define-a-function/
and
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/cht_cli/functions/upload-external-dependencies/

### Review Guidelines
[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/includes/steps-define-a-function-import-export.yaml [hunk 1]
```diff
@@ -1,153 +1,103 @@
-title: Export Your Realm Application
+title: Pull Your Realm Application
 ref: export-your-realm-application
 content: |
   To create new function with :doc:`realm-cli
```
Comment (mongomoe) [inline source/includes/steps-define-a-function-import-export.yaml:4]:
```suggestion
  To create a new function with :doc:`realm-cli
```

I realize this isn't part of your PR, but this should prob be fixed.

File: source/includes/steps-define-a-function-import-export.yaml [hunk 2]
```diff
@@ -1,153 +1,103 @@
...
+  Functions in {+service+} always execute in the context of a specific
+  application user or as a :ref:`system user <system-user>` (which
+  bypasses rules). To configure the function's execution user, specify
+  the type of authentication that {+service-short+} should use:
+
+  - **System**
+    
+    To execute a function as a System user, use the following configuration:
+
+   .. code-block:: json
```
Comment (nlarew) [inline source/includes/steps-define-a-function-import-export.yaml:56]: nit: indentation is off for this code block

File: source/includes/steps-define-a-function-import-export.yaml [hunk 3]
```diff
@@ -1,153 +1,103 @@
...
+
+  Set User Authentication
+  =======================
+
+  Functions in {+service+} always execute in the context of a specific
+  application user or as a :ref:`system user <system-user>` (which
+  bypasses rules). To configure the function's execution user, specify
+  the type of authentication that {+service-short+} should use:
+
+  - **System**
```
Comment (nlarew) [inline source/includes/steps-define-a-function-import-export.yaml:52]: Love the bullets vs the old table!

File: source/includes/steps-define-a-function-import-export.yaml [hunk 4]
```diff
@@ -1,153 +1,103 @@
...
+  see :ref:`Function Configuration Files <appconfig-functions>`.
+
 ---
-title: Configure Function Execution Logs
-ref: configure-function-execution-logs
+title: Configure Your Function
+ref: configure-your-function
 content: |
+
+  Set User Authentication
```
Comment (nlarew) [inline source/includes/steps-define-a-function-import-export.yaml:44]: Thoughts on bumping these headings up as individual steps? I could see arguments both ways.

**Review by mongomoe — APPROVED** [review]
lgtm

File: source/includes/steps-define-a-function-import-export.yaml [hunk 5]
```diff
@@ -195,23 +146,25 @@ content: |
   write the actual javascript code that runs when you :doc:`call the
   function </functions/call-a-function>`.
 
-  Add a file named ``source.js`` that contains the javascript code for
-  the function to the new function directory.
+  Create a subdirectory whose name matches the name of the function you 
+  specified in the configuration file, and then add a file named ``source.js``
+  to the new function directory. This file contains the javascript code for the 
+  function.
```
Comment (nlarew) [inline source/includes/steps-define-a-function-import-export.yaml:152]: `<function name>/source.js` is the old format - now it's `functions/**/<function name>.js`

File: source/includes/steps-define-a-function-import-export.yaml [hunk 6]
```diff
@@ -195,23 +146,25 @@ content: |
...
-title: Import the Function
+title: Push the Function
 ref: import-the-function
 content: |
   Once you've configured the function and written the source code, all
-  that's left is to import the function.
+  that's left is to push the function to {+service-short+}.
 
-  A. Ensure that ``config.json`` and ``source.js`` are saved.
+  A. Ensure that your changes to ``config.json`` and ``source.js`` are saved.
```
Comment (nlarew) [inline source/includes/steps-define-a-function-import-export.yaml:165]: - Need to update `source.js` -> `<function name>.js` here too
- Thoughts on framing this around "deploy the function" + removing the nested list? [example](https://github.com/mongodb/docs-realm/blob/cli/source/includes/steps-enforce-a-document-schema-import-export.yaml#L110-L115)

**Review by nlarew — CHANGES_REQUESTED** [review]
Overall looks good, just a few small changes & comments. Also they're not in the diff, but let's update the CLI Version note at the start of each Realm CLI procedure tab to indicate the procedure is updated for v2 (e.g. https://github.com/mongodb/docs-realm/blob/cli/source/mongodb/enforce-a-document-schema.txt#L44)

File: source/includes/steps-define-a-function-import-export.yaml [hunk 7]
```diff
@@ -1,153 +1,103 @@
...
+title: Pull Your Realm Application
 ref: export-your-realm-application
 content: |
   To create new function with :doc:`realm-cli
   </deploy/realm-cli-reference>`, you need a previously created
   :ref:`application configuration <application-directory>`.
-  
-  You can :doc:`export </deploy/export-realm-app>` your application from the
+
+  You can :doc:`pull </deploy/export-realm-app>` your application from the
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/includes/steps-define-a-function-import-export.yaml:8]: image

File: source/includes/steps-define-a-function-import-export.yaml (hunk 4 above)
Comment (MongoCaleb) [reply source/includes/steps-define-a-function-import-export.yaml:44]: They were, and I didn't like the way it scanned.

File: source/includes/steps-define-a-function-import-export.yaml (hunk 6 above)
Comment (MongoCaleb) [reply source/includes/steps-define-a-function-import-export.yaml:165]: image

File: source/includes/steps-define-a-function-import-export.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (nlarew) [reply source/includes/steps-define-a-function-import-export.yaml:8]: nit: let's change **Import/Export App** to just **Export App** to match the UI when this goes live

File: source/includes/steps-define-a-function-import-export.yaml (hunk 5 above)
Comment (nlarew) [reply source/includes/steps-define-a-function-import-export.yaml:152]: nit: a couple tweaks on this, basically to match the app structure page

* adds config.json
* fixes pipe character styling

```
app/
└── functions/
    ├── config.json
    ├── <function name>.js
    └── <directory>/
        └── <function name>.js


```

**Review by nlarew — APPROVED** [review]
LGTM % two more nits

**Pull request merged**